### PR TITLE
docs: Add initial date range to interval selection example

### PR DESCRIPTION
**Pull request by dsmedia**
This is a partial implementation of #3643, demonstrating how to set initial date ranges in interval selections through a concrete example. This improvement was made possible by the recent datetime support added in #3653 and #3662 by @dangotbanned.

Rather than creating a separate example, I opted to modify the existing interval selection example because:

1. Setting initial ranges is a core feature of interval selections, not a separate use case. Including it in the primary example helps users discover this capability alongside basic usage.

2. The change adds minimal complexity while showing the idiomatic way to set initial values using native datetime objects.

3. A single progressive example provides a clearer learning path than splitting this into two separate examples.

Changes made:
- Added demonstration of initial date range setting using datetime objects
- Updated example title and description to reflect the addition
- Added clarifying comments about the selection range

Remaining work in #3643:
- Add information about initial ranges to the selection documentation
- Consider documenting datetime handling in times_and_dates section

### CODE REVIEW

**Comment by dangotbanned** [comment]
@dsmedia thanks for opening this PR!

> Rather than creating a separate example, I opted to modify the existing interval selection example because:
> 
> 1. Setting initial ranges is a core feature of interval selections, not a separate use case. Including it in the primary example helps users discover this capability alongside basic usage.
> 2. The change adds minimal complexity while showing the idiomatic way to set initial values using native datetime objects.
> 3. A single progressive example provides a clearer learning path than splitting this into two separate examples.

I agree with these points. 
It makes more sense to me now that setting the value is simpler than the [stackoverflow](https://stackoverflow.com/questions/61469922/how-to-have-init-values-for-a-selection-interval-in-altair-when-using-dates/61470024#61470024) solution.

---

One question, do you think this example would make more sense in a different category?
I've noticed these two which have some similarities:

- https://altair-viz.github.io/gallery/select_mark_area.html
- https://altair-viz.github.io/gallery/interval_selection_map_quakes.html

But they are catgorised as [interactive charts](https://altair-viz.github.io/gallery/index.html#interactive-charts). 
This one is under [area charts](https://altair-viz.github.io/gallery/index.html#area-charts)

There is also [VL-interactive_overview_detail](https://vega.github.io/vega-lite/examples/interactive_overview_detail.html)

I know that this wasn't a change that you introduced, but I think it would further improve discoverability of this functionality

File: tests/examples_arguments_syntax/interval_selection.py [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
-==================
+Interval Selection with Initial Date Range
+==========================================
 
 This is an example of creating a stacked chart for which the domain of the
-top chart can be selected by interacting with the bottom chart.
+top chart can be selected by interacting with the bottom chart. The initial
+selection range is set using Python's native datetime objects.
 """
 # category: area charts
```
Comment (dangotbanned) [inline tests/examples_arguments_syntax/interval_selection.py:9]:
```suggestion
# category: interactive charts
```

File: tests/examples_methods_syntax/interval_selection.py [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
-==================
+Interval Selection with Initial Date Range
+==========================================
 
 This is an example of creating a stacked chart for which the domain of the
-top chart can be selected by interacting with the bottom chart.
+top chart can be selected by interacting with the bottom chart. The initial
+selection range is set using Python's native datetime objects.
 """
 # category: area charts
```
Comment (dangotbanned) [inline tests/examples_methods_syntax/interval_selection.py:9]:
```suggestion
# category: interactive charts
```

**Review by dangotbanned — APPROVED** [review]
Thanks again for the PR @dsmedia 

I'm approving now, will leave open for a bit to see if anyone else wants to weigh in on the [suggestions](https://github.com/vega/altair/pull/3667#issuecomment-2450249670)

**Comment by mattijn** [comment]
Applied suggestions from review and activated auto-merge. Thanks @dsmedia! Much appreciated!

**Review by mattijn — APPROVED** [review]

**Pull request merged**